### DISCUSSION TITLE
how to add fragment anchor to header?

### DISCUSSION THREAD

**Discussion by bluebrown** [yuin/goldmark#355] [Q&A]
Hi, I would like to have some kind of link in the headers so that it will be easy to share a relevant section of the document.

There is an extension that generates Ids for headers but the Id alone is not all that useful. In order for it to become useful, there should be an anchors referencing to that section, preferable inside the header or wrapping it.

I could not find out how do achieve this. I think its a fairly common need and I am properly missing something in the docs.

**Comment by abhinav** [comment]
@bluebrown if an extra dependency is okay, I've extracted and generalized the code I use for this into an extension: [goldmark-anchor](https://github.com/abhinav/goldmark-anchor). 

To use it, just add `anchor.Extender` to your goldmark extensions.
You still need to have WithAutoHeadingID enabled to get goldmark to generate the IDs.

```go
goldmark.New(
  goldmark.WithParserOptions(
    parser.WithAutoHeadingID(),
  ),
  goldmark.WithExtensions(
    // ...
    &anchor.Extender{},
  ),
).Convert(src, out)
```

The README has more instructions.

**Reply by bluebrown** [reply]
I will check it out. Thank you
